edgepi-digital-out.py

```diff
@@ -6,27 +6,31 @@
 _logger=logging.getLogger(__name__)
 logging.basicConfig(level=logging.INFO)
 
-# arv[0] is this file's name, argv[1] should be numeric command
-if len(sys.argv) > 1:
+# create new DOUT instance
+try:
+    digital_output = EdgePiDigitalOutput()
+except ModuleNotFoundError as e:
+    _logger.error(f'Failed to load EdgePi modules: {e}')
+    sys.exit(0)
+
+# node channel property to SDK enum
+channels = {"dout1": GpioPins.DOUT1, "dout2": GpioPins.DOUT2, "dout3": GpioPins.DOUT3, "dout4": GpioPins.DOUT4,
+"dout5": GpioPins.DOUT5, "dout6":GpioPins.DOUT6, "dout7":GpioPins.DOUT7, "dout8": GpioPins.DOUT8}
+
+# keep receiving commands from parent process
+while True:
     try:
-        digital_output = EdgePiDigitalOutput()
-    except ModuleNotFoundError as e:
-        _logger.error(f'Failed to load EdgePi modules: {e}')
-        sys.exit(0)
-    
-    # keep receiving commands from parent process
-    while True:
-        try:
-            cmd = input()
-            # exit signal from parent process or user exit from child process
-            if 'exit' in cmd or cmd == '0':
-                sys.exit(0)
-            elif cmd == '5':
-                digital_output.digital_output_direction(GpioPins.DOUT4, False)
-                digital_output.digital_output_state(GpioPins.DOUT4, False)
-                print("Hello")
-        except(EOFError, SystemExit, KeyboardInterrupt):
-            sys.exit(0)
-else:
-    print("Number of commands must equal 2")
-    
\ No newline at end of file
+        # parse input from js file
+        inputs = input().split(",")
+        # get channel enum
+        channel = channels[inputs[0]]
+        # get on/off
+        on_off = True if inputs[1].strip() == "true" else False
+
+        # action on dout based on input gathered
+        digital_output.digital_output_direction(channel, False)
```

Comment:
if you know the channel you are using is always an output, `digital_output_direction()` can be called only once before the while loop starts.

Reply:
The loop is there to continually receive input from the js file. I could set the direction for all output pins to False before the while loop but that might mess up what users want to use the other pins for. This way it only sets the direction for the pin that the user asks for. It is redundant because it will keep setting the direction and state every time the node receives a message... I'll look into it